File: update_compras.py
import re
with open('/tmp/compras.tsx', 'r') as f:
    content = f.read()

# 1. Add fornecedores state
content = content.replace("const [categorias, setCategorias] = useState<any[]>([]);", 
"const [categorias, setCategorias] = useState<any[]>([]);\n  const [fornecedores, setFornecedores] = useState<any[]>([]);\n  const [showFornecedorModal, setShowFornecedorModal] = useState(false);\n  const [novoFornecedorNome, setNovoFornecedorNome] = useState('');")

# 2. Add fornecedor_id to compraForm
content = content.replace("fornecedor: '',", "fornecedor: '',\n    fornecedor_id: '',")

# 3. Add getFornecedores to fetchData
content = content.replace("api.getMaterialCategories()", "api.getMaterialCategories(),\n        api.getFornecedores({ ativo: true })")
content = content.replace("const [comprasData, obrasData, materiaisData, categoriasData] = await Promise.all([", "const [comprasData, obrasData, materiaisData, categoriasData, fornecedoresData] = await Promise.all([")
content = content.replace("setCategorias(categoriasData);", "setCategorias(categoriasData);\n      setFornecedores(fornecedoresData);")

# 4. handleOpenNew update
content = content.replace("fornecedor: '',", "fornecedor: '',\n      fornecedor_id: '',")

# 5. UI change for fornecedor input (Combobox / Select with Plus button)
ui_search = """          <div>
            <label className="block text-sm font-medium text-gray-700 mb-1">Fornecedor</label>
            <input
              type="text"
              placeholder="Nome do fornecedor ou loja"
              value={compraForm.fornecedor}
              onChange={e => setCompraForm({...compraForm, fornecedor: e.target.value})}
              className="w-full rounded-lg border border-gray-300 px-3 py-2 focus:ring-blue-500 focus:border-blue-500"
            />
          </div>"""

ui_replace = """          <div>
            <label className="block text-sm font-medium text-gray-700 mb-1">Fornecedor</label>
            <div className="flex gap-2">
              <select
                value={compraForm.fornecedor_id}
                onChange={e => {
                  const f = fornecedores.find(x => x.id === e.target.value);
                  setCompraForm({ ...compraForm, fornecedor_id: e.target.value, fornecedor: f ? f.nome : '' });
                }}
                className="flex-1 rounded-lg border border-gray-300 px-3 py-2 focus:ring-blue-500 focus:border-blue-500"
              >
                <option value="">Selecione ou adicione...</option>
                {fornecedores.map(f => (
                  <option key={f.id} value={f.id}>{f.nome}</option>
                ))}
              </select>
              <button 
                type="button" 
                onClick={() => setShowFornecedorModal(true)}
                className="bg-gray-100 text-gray-700 p-2 rounded-lg hover:bg-gray-200 border border-gray-300"
                title="Cadastrar Novo Fornecedor"
              >
                <Plus className="w-5 h-5" />
              </button>
            </div>
          </div>"""

content = content.replace(ui_search, ui_replace)

# 6. Add quick create modal at the bottom
modal = """
      {/* Modal Novo Fornecedor Rápido */}
      {showFornecedorModal && (
        <div className="fixed inset-0 z-50 overflow-y-auto">
          <div className="flex items-center justify-center min-h-screen px-4 pt-4 pb-20 text-center sm:p-0">
            <div className="fixed inset-0 transition-opacity bg-gray-500 bg-opacity-75" onClick={() => setShowFornecedorModal(false)} />
            
            <div className="relative inline-block w-full max-w-md p-6 overflow-hidden text-left align-middle transition-all transform bg-white shadow-xl rounded-2xl">
              <h3 className="text-lg font-bold text-gray-900 mb-4">Novo Fornecedor Rápido</h3>
              <div className="space-y-4">
                <div>
                  <label className="block text-sm font-medium text-gray-700 mb-1">Nome do Fornecedor *</label>
                  <input
                    type="text"
                    autoFocus
                    className="w-full border-gray-300 rounded-lg focus:ring-blue-500 focus:border-blue-500"
                    value={novoFornecedorNome}
                    onChange={(e) => setNovoFornecedorNome(e.target.value)}
                    placeholder="Ex: Casa Souza"
                  />
                </div>
                <div className="flex justify-end gap-3 pt-4 border-t border-gray-100">
                  <button
                    type="button"
                    onClick={() => setShowFornecedorModal(false)}
                    className="px-4 py-2 border border-gray-300 rounded-lg text-gray-700 font-medium hover:bg-gray-50"
                  >
                    Cancelar
                  </button>
                  <button
                    type="button"
                    onClick={async () => {
                      if (!novoFornecedorNome.trim()) return;
                      try {
                        const newF = await api.createFornecedor({ nome: novoFornecedorNome.trim() });
                        setFornecedores([...fornecedores, newF].sort((a,b) => a.nome.localeCompare(b.nome)));
                        setCompraForm({ ...compraForm, fornecedor_id: newF.id, fornecedor: newF.nome });
                        setShowFornecedorModal(false);
                        setNovoFornecedorNome('');
                      } catch(e: any) {
                        alert(e.message);
                      }
                    }}
                    className="bg-blue-600 text-white px-4 py-2 rounded-lg font-medium hover:bg-blue-700"
                  >
                    Salvar
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      )}
"""
content = content.replace("    </div>\n  );\n}", modal + "\n    </div>\n  );\n}")

with open('/tmp/compras.tsx', 'w') as f:
    f.write(content)

File: update_compras.js
import fs from 'fs';
let content = fs.readFileSync('src/pages/admin/ComprasMateriaisTab.tsx', 'utf-8');

// 1. Import apiMateriaisRPC
content = content.replace(
  `import { format } from 'date-fns';`,
  `import { format } from 'date-fns';\nimport { apiMateriaisRPC } from '../../lib/api-materiais';`
);

// 2. Add handler for Fornecedor modal
const fornecedorHandler = `
  const handleCadastrarFornecedor = async () => {
    if (!novoFornecedorNome.trim()) {
      alert('Informe o nome do fornecedor');
      return;
    }
    try {
      setIsSaving(true);
      const novoId = await apiMateriaisRPC.cadastrarFornecedorRapido({ p_nome: novoFornecedorNome.trim() });
      setFormSuccess('Fornecedor cadastrado com sucesso!');
      setShowFornecedorModal(false);
      setNovoFornecedorNome('');
      
      // Refresh list
      const fornecedoresData = await api.getFornecedores({ ativo: true });
      setFornecedores(fornecedoresData);
      
      // Auto-select
      if (novoId) {
        setCompraForm(prev => ({ ...prev, fornecedor_id: novoId }));
      }
    } catch (err: any) {
      if (err.message && err.message.includes('já está cadastrado')) {
        alert('Este fornecedor já está cadastrado.');
      } else {
        alert('Erro ao cadastrar fornecedor.');
      }
    } finally {
      setIsSaving(false);
    }
  };
`;
// Let's insert the handler before handleSave
content = content.replace(
  `  const handleSave = async (e: React.FormEvent) => {`,
  fornecedorHandler + `\n  const handleSave = async (e: React.FormEvent) => {`
);

// 3. Rewrite handleSave to use apiMateriaisRPC
const handleSaveNew = `  const handleSave = async (e: React.FormEvent) => {
    e.preventDefault();
    setFormError('');
    setFormSuccess('');

    if (!compraForm.data_compra) return setFormError('A data da compra é obrigatória.');
    if (!compraForm.obra_id) return setFormError('A obra é obrigatória.');
    if (!compraForm.fornecedor_id) return setFormError('O fornecedor é obrigatório.');
    if (itensForm.length === 0) return setFormError('Adicione pelo menos um item à compra.');

    const itensValidos = [];
    for (let i = 0; i < itensForm.length; i++) {
      const item = itensForm[i];
      if (!item.material_id) return setFormError(\`Selecione o produto para o item \${i + 1}.\`);
      if (item.quantidade <= 0) return setFormError(\`A quantidade do item \${i + 1} deve ser maior que zero.\`);
      if (item.valor_unitario < 0) return setFormError(\`O valor unitário do item \${i + 1} não pode ser negativo.\`);
      
      itensValidos.push({
        material_id: item.material_id,
        quantidade: item.quantidade,
        valor_unitario: item.valor_unitario
      });
    }

    try {
      setIsSaving(true);
      await apiMateriaisRPC.registrarCompraMaterial({
        p_data_compra: compraForm.data_compra,
        p_fornecedor_id: compraForm.fornecedor_id,
        p_numero_recibo: compraForm.numero_recibo || null,
        p_obra_id: compraForm.obra_id,
        p_itens: itensValidos
      });
      setFormSuccess('Compra registrada com sucesso!');
      await fetchData();
      setTimeout(() => { setView('list'); }, 1500);
    } catch (err: any) {
      setFormError(err.message || 'Erro ao registrar compra.');
    } finally {
      setIsSaving(false);
    }
  };`;

// replace old handleSave
content = content.replace(
  /const handleSave = async \(e: React.FormEvent\) => \{[\s\S]*?catch \(err: any\) \{[\s\S]*?finally \{[\s\S]*?\}\n  \};/,
  handleSaveNew
);

// 4. Update the Obra selector logic to show hierarchy
const mainObras = `obras.filter(o => !o.parent_obra_id)`;
const obraOptions = `{obras.filter(o => !o.parent_obra_id).map(obra => (
                <optgroup key={obra.id} label={obra.nome}>
                  <option value={obra.id}>{obra.nome} (Principal)</option>
                  {obras.filter(sub => sub.parent_obra_id === obra.id).map(sub => (
                    <option key={sub.id} value={sub.id}>-- {sub.nome}</option>
                  ))}
                </optgroup>
              ))}`;

content = content.replace(
  /\{obras.map\(obra => \(\s*<option key=\{obra.id\} value=\{obra.id\}>\{obra.nome\}<\/option>\s*\)\)\}/,
  obraOptions
);

// 5. Date field logic: Disable for Operador
const isAdminDecl = `const isAdmin = usuario?.perfil === 'ADMIN';`;
content = content.replace(
  `const { usuario } = useAuth();`,
  `const { usuario } = useAuth();\n  const isAdmin = usuario?.perfil === 'ADMIN';`
);

content = content.replace(
  /<input\s*type="date"\s*required\s*value=\{compraForm.data_compra\}[\s\S]*?\/>/,
  `<input
              type="date"
              required
              disabled={!isAdmin}
              value={compraForm.data_compra}
              onChange={e => setCompraForm({...compraForm, data_compra: e.target.value})}
              className="w-full rounded-lg border border-gray-300 px-3 py-2 focus:ring-blue-500 focus:border-blue-500 disabled:bg-gray-100 disabled:text-gray-500"
            />`
);

// 6. Remove Observação field
content = content.replace(
  /<div className="md:col-span-2">\s*<label className="block text-sm font-medium text-gray-700 mb-1">Observação<\/label>[\s\S]*?<\/textarea>\s*<\/div>/,
  ``
);

// 7. Render CADASTRO FORNECEDOR modal
const modalJSX = `
      {showFornecedorModal && (
        <div className="fixed inset-0 bg-black bg-opacity-50 flex items-center justify-center z-50 p-4">
          <div className="bg-white rounded-xl shadow-xl w-full max-w-md overflow-hidden">
            <div className="p-4 border-b border-gray-200 flex justify-between items-center bg-gray-50">
              <h3 className="text-lg font-bold text-gray-900">Cadastro Rápido de Fornecedor</h3>
              <button onClick={() => setShowFornecedorModal(false)} className="text-gray-500 hover:text-gray-700">
                <X className="w-5 h-5" />
              </button>
            </div>
            <div className="p-6">
              <label className="block text-sm font-medium text-gray-700 mb-1">Nome do fornecedor *</label>
              <input
                type="text"
                autoFocus
                value={novoFornecedorNome}
                onChange={e => setNovoFornecedorNome(e.target.value)}
                className="w-full rounded-lg border border-gray-300 px-3 py-2 focus:ring-blue-500 focus:border-blue-500 mb-4"
              />
              <div className="flex gap-3 justify-end">
                <button
                  type="button"
                  onClick={() => setShowFornecedorModal(false)}
                  className="px-4 py-2 border border-gray-300 text-gray-700 rounded-lg hover:bg-gray-50"
                  disabled={isSaving}
                >
                  Cancelar
                </button>
                <button
                  type="button"
                  onClick={handleCadastrarFornecedor}
                  className="px-4 py-2 bg-blue-600 text-white rounded-lg hover:bg-blue-700 disabled:opacity-50"
                  disabled={isSaving}
                >
                  {isSaving ? 'Salvando...' : 'Cadastrar'}
                </button>
              </div>
            </div>
          </div>
        </div>
      )}
`;

content = content.replace(
  /return \(\s*<div/,
  modalJSX + `\n  return (\n    <div`
);

fs.writeFileSync('src/pages/admin/ComprasMateriaisTab.tsx', content);
